### PR TITLE
mark SBU download as not accessible in tests

### PR DESCRIPTION
**Pull request by pmeier**
Addresses #7964. Since final RC for upcoming release is already cut, we can just leave it in the "not accessible" list until next release. Maybe it comes up again.

### CODE REVIEW

File: test/test_datasets_download.py [hunk 1]
```diff
@@ -391,4 +391,4 @@ def test_url_is_not_accessible(url):
     If you see this test failing, find the offending dataset in the parametrization and move it to
     ``test_url_is_accessible``.
     """
-    retry(lambda: assert_url_is_accessible(url))
+    assert_url_is_accessible(url)
```
Comment (pmeier) [inline test/test_datasets_download.py:394]: Driveby. Since we assert that the URL is not accessible by trying and xfailing afterwards, there is no need for retries.

**Review by vfdev-5 — APPROVED** [review]
Stamping, thanks @pmeier !
